### PR TITLE
feat: added optional filtering by shard to selectPeer()

### PR DESCRIPTION
**Pull request by SionoiS**
# Description
Tiny PR adding an optional shard filter when selecting a peer via peer manager.

Tracking #1941

### CODE REVIEW

File: waku/node/peer_manager/peer_manager.nim [hunk 1]
```diff
@@ -631,11 +632,14 @@ proc prunePeerStore*(pm: PeerManager) =
                                        capacity = capacity,
                                        pruned = pruned
 
-proc selectPeer*(pm: PeerManager, proto: string): Option[RemotePeerInfo] =
+proc selectPeer*(pm: PeerManager, proto: string, shard: Option[PubsubTopic] = none(PubsubTopic)): Option[RemotePeerInfo] =
```
Comment (jm-clius) [inline waku/node/peer_manager/peer_manager.nim:635]: In the spirit of `shard` being the number encoded into the `pubsubTopic`, could we call this something like `shardedPubsubTopic`?

**Review by jm-clius — APPROVED** [review]
LGTM for now, but wondering if we'll always have the ENR for interesting peers available? The idea of some metadata exchange protocol, for example, would allow us to discover the shards for any peer even outside of ENR encoding. In other words, we may want to also index against the shard, cluster in the peerStore so that it's not a separate filter loop.

**Comment by SionoiS** [comment]
> LGTM for now, but wondering if we'll always have the ENR for interesting peers available? The idea of some metadata exchange protocol, for example, would allow us to discover the shards for any peer even outside of ENR encoding. In other words, we may want to also index against the shard, cluster in the peerStore so that it's not a separate filter loop.

I agree that sharding is not well integrated into peer management yet. I don't think that ENRs will be missing in many cases but we def should have something in place for those.

File: waku/node/peer_manager/peer_manager.nim (hunk 1 above)
Comment (SionoiS) [reply waku/node/peer_manager/peer_manager.nim:635]: > In the spirit of `shard` being the number encoded into the `pubsubTopic`, 

Is it? I always thought of shards as pubsub topic that conform to static or autoshard specs. The numbers are named cluster and index.

> could we call this something like `shardedPubsubTopic`?
The term `shardedPubsubTopic` has never been used before. I don't like adding more terminology.

**Pull request merged**